### DISCUSSION TITLE
短期开发计划 （ 2024 年 2 月）

### DISCUSSION THREAD

**Discussion by cloudwu** [ejoy/ant#78] [Announcements]
* [x] 把我们正在制作的游戏开源。一个完整的游戏的代码和相关美术资产有较大的参考价值。
* [x] 完善 vfs 仓库的打包发布、补丁下载。
* [ ] RmlUI 完善，特别是 CSS 驱动的动画。
* [ ] RmlUI 中使用的图片整合到一张大贴图中，支持 Atlas 描述图集。如果将来支持 2d 渲染管线，会将复用图集的功能。(延后）
* [x] RmlUI 中的九宫格背景图片配置在 .texture 中，不需要在 css 里额外描述。
* [x] 更好的阴影效果 ，完善  vms 软阴影
* [x] 优化相同物件合并渲染
* [x] 修复点光源（在我们目前的游戏中没用到）因近期引擎重构引入的 bug 。
* [ ] 修复编辑器中过去不常用功能因近期引擎重构引入 bug 。（正在进行中）
* [ ] 重构后处理中的抗锯齿系统 TAA 。(延后)

未完成计划移到 3 月 ：https://github.com/ejoy/ant/discussions/106

**Comment by julywind168** [comment]
RmiUI 这块是得好好完善
- 好像 input 无法输入
- 好像所有元素都是块级元素 (一个 div 内放2 span 都会换行)
- 会支持 react/vue 那样的组件化开发吗, 我正在开发 mmo, UI 页面可能会相当复杂

**Reply by cloudwu** [reply]
我们在做手机游戏，手机上 input 控件现在没实现。手机上应该会更麻烦一些。

我们现在 UI 也挺复杂的，我版面复杂觉得不是什么问题。这个是严格遵循 css 标准的，所以版面和期望不符合，要么是 bug 需要改，要么是用错了？

**Reply by julywind168** [reply]
是我 css 理解不到位了 = =!

**Comment by shar0** [comment]
虽然感觉现阶段并不重要，是否可在 Editor 里集成几个简易的 “项目管理” 功能呢？
* 创建项目：设置项目目录，自动生成 Hello World Pkg
* 打开项目：通过一个类似 *.antproject 保存项目配置，可以用 Editor 打开
* 运行项目：在 Editor 中增加一个 “Play” 或 “luamake + run” 的功能

至于剩下的如 “历史项目” 等更多功能，暂时看起来也不太需要。

**Reply by cloudwu** [reply]
“项目” 我建议和一般软件项目一样管理：依靠 git 建立工作流，而不是依赖编辑器建立工作流。

在编辑器中做多项目管理，我觉得意义很小。一般的正式开发中，同时只会有一个项目，不会在多个项目中切来切去。即使实现 “项目”，也应该是在启动命令行传入，并可以设为默认值。需要在很多项目间切换，我觉得更多满足的是：“玩玩” ，而不是“做事” 。

**Reply by cloudwu** [reply]
另外，项目会和 ant 的库 link 在一起（因为 ios 不支持动态库）。所以，项目本身一般会有一个独立的启动程序。这个程序其实还是一个 lua 解释器，但可以在这个程序中做一些扩展：支持默认参数等，就不需要在命令行加参数了。这样，在开发机上，双击它就能打开游戏，不依赖控制台。

过完年，会把我们的游戏开源，可以看看我们游戏的启动程序怎么做的。

**Reply by shar0** [reply]
了解，不过在多项目中切换，特别是在游戏开发中，是有实际应用场景的。
如小游戏、超休闲类游戏开发周期中，存在大量快速开项目出品（3个月一款，甚至1个月），并存在多项目中复用模块的情况（如广告、游戏设置、启动屏）。
因而既有的堆积的项目，由于开发节奏快、堆积项目多，不能模块化的东西多，旧有项目也会经常反复点开作为对照。
这块最卷的代表可能是Voodoo Games，会把引擎的所有功能都用于 “创作”（或是说整活儿）。
这个常见于移动端，而且也可以靠命令行指令脚手架生成项目模板。
不过每个引擎都有自己擅长的领域就是了。

**Comment by terry8210** [comment]
如果skynet也能把一个游戏的服务器开源出来做为一个标准，现在100个人，有100个人使用skynet的办法。
